orchagent/main.cpp

```diff
@@ -286,6 +413,43 @@ int main(int argc, char **argv)
         attrs.push_back(attr);
     }
 
+    // Get info required for VOQ system and connect to CHASSISS_APP_DB
+    shared_ptr<DBConnector> chassis_app_db;
+    vector<sai_system_port_config_t> sysportconfiglist;
+    if ((gMySwitchType == "voq") &&
+            (getSystemPortConfigList(&config_db, &appl_db, sysportconfiglist)))
+    {
+        attr.id = SAI_SWITCH_ATTR_TYPE;
+        attr.value.u32 = SAI_SWITCH_TYPE_VOQ;
+        attrs.push_back(attr);
+
+        attr.id = SAI_SWITCH_ATTR_SWITCH_ID;
+        attr.value.u32 = gVoqMySwitchId;
+        attrs.push_back(attr);
+
+        attr.id = SAI_SWITCH_ATTR_MAX_SYSTEM_CORES;
+        attr.value.u32 = gVoqMaxCores;
+        attrs.push_back(attr);
+
+        gCfgSystemPorts = (uint32_t) sysportconfiglist.size();
+        if (gCfgSystemPorts)
```

Comment:
assert that this is > 0  ?

Reply:
Yes. We'll fix this

Reply:
Fixed. Instead of using assert, I log error and exit failure